var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -57,178 +50,90 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
         depends_on(
-            "legion amdgpu_target=" + _flag,
-            when="backend=legion +rocm amdgpu_target=" + _flag + " @2.0:",
+            f"legion amdgpu_target={_flag}", when=f"backend=legion +rocm amdgpu_target={_flag}"
         )
 
-    requires("%gcc@9:", when="@2: %gcc", msg="Version 9 or newer of GNU compilers required!")
-
-    conflicts("+tutorial", when="backend=hpx")
-    # FleCSI@2: no longer supports serial or charmpp backends
-    conflicts("backend=serial", when="@2.0:")
-    conflicts("backend=charmpp", when="@2.0:")
-    # FleCSI@:1.4 releases do not support kokkos, omp, cuda, or rocm
-    conflicts("+kokkos", when="@:1.4")
-    conflicts("+openmp", when="@:1.4")
-    conflicts("+cuda", when="@:1.4")
-    conflicts("+rocm", when="@:1.4")
-    # Unit tests require flog support
-    conflicts("+unit_tests", when="~flog")
+    requires("%gcc@9:", when="%gcc", msg="Version 9 or newer of GNU compilers required!")
+
     # Disallow conduit=none when using legion as a backend
     conflicts("^legion conduit=none", when="backend=legion")
-    # Due to overhauls of Legion and Gasnet spackages
-    #   flecsi@:1.4 can no longer be built with a usable legion
-    conflicts("backend=legion", when="@:1.4")
-    conflicts("+hdf5", when="@2: backend=hpx", msg="HPX backend doesn't support HDF5")
+    conflicts("+hdf5", when="backend=hpx", msg="HPX backend doesn't support HDF5")
 
     def cmake_args(self):
         spec = self.spec
 
-        if spec.satisfies("@2.2:"):
-            options = [
-                self.define_from_variant("FLECSI_BACKEND", "backend"),
-                self.define_from_variant("CALIPER_DETAIL", "caliper_detail"),
-                self.define_from_variant("ENABLE_FLOG", "flog"),
-                self.define_from_variant("ENABLE_GRAPHVIZ", "graphviz"),
-                self.define_from_variant("ENABLE_HDF5", "hdf5"),
-                self.define_from_variant("ENABLE_KOKKOS", "kokkos"),
-                self.define_from_variant("ENABLE_OPENMP", "openmp"),
-                self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
-                self.define("ENABLE_UNIT_TESTS", self.run_tests),
-                self.define_from_variant("ENABLE_DOCUMENTATION", "doc"),
-            ]
-
-            if "+rocm" in self.spec:
-                options.append(self.define("CMAKE_CXX_COMPILER", self.spec["hip"].hipcc))
-                options.append(self.define("CMAKE_C_COMPILER", self.spec["hip"].hipcc))
-                if "backend=legion" in self.spec:
-                    # CMake pulled in via find_package(Legion) won't work without this
-                    options.append(self.define("HIP_PATH", "{0}/hip".format(spec["hip"].prefix)))
-            elif "+kokkos" in self.spec:
-                options.append(self.define("CMAKE_CXX_COMPILER", self.spec["kokkos"].kokkos_cxx))
-        else:
-            # kept for supporing version prior to 2.2
```

Comment:
We're not removing the 2.0/2.1 versions; is it OK to simplify this part this much?

Reply:
No, you're right. That was too aggressive.